utility/src/main/java/org/oppia/android/util/logging/CurrentAppScreenNameWrapper.kt

```diff
@@ -0,0 +1,24 @@
+package org.oppia.android.util.logging
+
+import android.content.Intent
+import org.oppia.android.app.model.CurrentAppScreen
+import org.oppia.android.app.model.ScreenName
+import org.oppia.android.util.extensions.putProtoExtra
+import javax.inject.Inject
+
+private const val CURRENT_APP_SCREEN_NAME_KEY = "key"
+
+/** Utility to provide utilities related to wrapping the current app-screen name in an intent. */
+class CurrentAppScreenNameWrapper @Inject constructor() {
+
+  /** Returns an intent packed with [CurrentAppScreen]. */
+  fun getCurrentAppScreenNameIntent(screenName: ScreenName): Intent = Intent().apply {
```

Comment:
Per earlier suggestions, this could be: decorateWithScreenName

Reply:
done.